libs/barista-components/filter-field/README.md

```diff
@@ -151,9 +154,14 @@ interface.
 
 The filter-field provides a default implementation of a `DataSource`, named
 `DtFilterFieldDefaultDataSource`, that takes a specific form of data
-(`DtFilterFieldDefaultDataSourceType`). If your data has a different form you
-can either transform your data so the default data source can understand it, or
-you create your own custom data source.
+(`DtFilterFieldDefaultDataSourceType`). Theoretically, the filter-field supports
+an infinite nesting of nodes. Only special nodes like range and distinct need to
+be leaf nodes (the last node of a branch).
+
+<ba-live-example name="DtExampleFilterFieldInfiniteDataDepth"></ba-live-example>
+
+If your data has a different structure you can either transform it to fit the
+default data source structure, or create your own custom data source.
```

Comment:
NIT: I would not state that you should transform it – I would state that you should create your own data source